File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        ios_base::sync_with_stdio(0);
        cin.tie(0);
        cout.tie(0);

        int n = nums.size(), idx = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] != 0) {
                int temp = nums[idx];
                nums[idx] = nums[i];
                nums[i] = temp;

                idx++;
            }
        }
    }
};

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int n = nums.length, i = 0;

        // for (int num: nums) {
        //     if (num != 0) {
        //         nums[i++] = num;
        //     }
        // }

        // while (i < n) {
        //     nums[i++] = 0;
        // }

        for (int j = 0; j < n; j++) {
            if (nums[j] != 0) {
                int t = nums[i];
                nums[i] = nums[j];
                nums[j] = t;
                i++;
            }
        }

    }
}